### ISSUE TITLE
Document that nested `Run Keyword If` with `ELSE/ELSE IF` is not supported

### ISSUE THREAD

**Issue by adam-wypych** [robotframework/robotframework#2871]
Hi,

I've been tried to find an information if syntax like: 
```robotframework
Run Keyword If  (condition_1) 
            Run Keyword If  (condition_2) (new_keyword_1) ELSE (else_keyword_1)
ELSE
             (else_keyword_2)
```
is supported. So far I only I've found only an information on stackoverflow https://stackoverflow.com/questions/34590001/robot-framework-nested-if-statement  saying "strickly speaking, no".

What I observed is that RF:
a) do not inform that syntax is not supported
b) ELSE for `(condition_2)` is used for both `(condition_1)` and `(condition_2)` 

If the syntax is not supported than I think an error message should be printed or keyword should failed.

**Comment by pekkaklarck** [comment]
Nested `Run Keyword If` usage is not supported. It would be hard to parse such syntax and know which `ELSE` or `ELSE IF` belongs to which keyword. Also, I don't see much usages for such functionality when you can have `ELSE IF` branches. If such structure is needed, a workaround is using the nested `Run Keyword If` in a keyword used by the top-level `Run Keyword If`.

I don't think it is worth the effort to try detecting possible nested `Run Keyword If` usage and make it an error. It would even be a backwards-incompatible change preventing passing `Run Keyword If` sting as an argument. We could easily document that nested used is not supported in `Run Keyword If` documentation.

**Comment by adam-wypych** [comment]
@pekkaklarck Thanks Pekka, this is kind of the answer, which I expected.
